sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1761,4 +1763,67 @@ private static void checkEmptyStringAllowed(Class<?> type, JavaType genericType,
     }
     return supportedRunners.build();
   }
+
+  /**
+   * @param properties the properties to use to create the pipeline options instance.
+   * @return a pipeline options instance based on the specified properties.
+   */
+  public static PipelineOptions fromProperties(final Properties properties) {
+    if (properties == null) {
+      return create();
+    }
+    final String[] args = properties.stringPropertyNames()
+      .stream()
+      .map(k -> String.format("--%s=%s", k, properties.getProperty(k)))
+      .toArray(String[]::new);
+    return fromArgs(args).create();
+  }
+
+  /**
+   * @param prefix the prefix filter applied on system properties keys.
+   * @param properties the properties to use to create the pipeline options instance.
+   * @return a pipeline options instance based on the specified properties
+   * filtered with the specified prefix.
+   */
+  public static PipelineOptions fromProperties(final String prefix, final Properties properties) {
+    if (prefix == null) {
+      return fromProperties(properties);
+    }
+    if (properties == null) {
+      return create();
+    }
+    final Properties instantiationProperties = properties.stringPropertyNames()
+      .stream()
+      .filter(k -> k.startsWith(prefix))
+      .collect(
+        Properties::new,
+        (p, k) -> p.setProperty(k.substring(prefix.length()), properties.getProperty(k)),
+        Hashtable::putAll);
+    return fromProperties(instantiationProperties);
+  }
+
+  /**
+   * @param prefix the prefix filter applied on system properties keys.
+   * @return a pipeline options instance based on current environment and overriden with
+   * system properties key/values and filtered with a custom prefix.
+   */
+  public static PipelineOptions fromJvm(final String prefix) {
+    final Properties properties = System.getenv()
+      .entrySet()
+      .stream()
+      .collect(
+              Properties::new,
+              (p, e) -> p.setProperty(e.getKey(), e.getKey()),
+              Properties::putAll);
+    properties.putAll(System.getProperties());
```

Comment:
Don't blend JVM properties with environment variables. For this feature, just use system properties. The way to interact with the environment is to put the whole JSON into one variable.

Reply:
This way is more convenient for a lot of people. I saw nobody to be able to use properly and reliably the JSON flavor cause depending where you set it the escaping is different (console, IDE, shell scripts, ...). Using the env enables ops to use that *naturally*, using the system props enables the dev to use it *naturally*. This is where it is coming from.

Reply:
This particular bit of the PR is not agreed on. I don't like it - it seems like a real abuse of env vars. There could be a zillion pipeline options and it is very weird to have a cluster of env vars parsed together like this. I could be OK with it but I want the dev list to discuss.

If you do want to also support this, do it in a separate PR, separate method, and get some feedback on the dev list about that.

Reply:
I ll ping today the list. If a discusdion is needed ill split it, ok?